inc/class-statify-frontend.php

```diff
@@ -420,32 +447,35 @@ public static function amp_post_template_analytics( $analytics ) {
 			$analytics['statify'] = array(
 				'type'        => '',
 				'attributes'  => array(),
-				'config_data' => array(
-					'extraUrlParams' => array(
-						'action'           => 'statify_track',
-						'_ajax_nonce'      => wp_create_nonce( 'statify_track' ),
-						'statify_referrer' => '${documentReferrer}',
-						'statify_target'   => '${canonicalPath}amp/',
-					),
-					'requests'       => array(
-						'event' => admin_url( 'admin-ajax.php' ),
-					),
-					'triggers'       => array(
-						'trackPageview' => array(
-							'on'      => 'visible',
-							'request' => 'event',
-							'vars'    => array(
-								'eventId' => 'pageview',
-							),
-						),
-					),
-					'transport'      => array(
-						'xhrpost' => true,
-					),
-				),
+				'config_data' => self::make_amp_config(),
 			);
 		}
 
 		return $analytics;
 	}
+
+	/**
+	 * Generate AMP-analytics configuration.
+	 *
+	 * @return array Configuration array.
+	 */
+	private static function make_amp_config() {
+		return array(
+			'requests'       => array(
+				'pageview' => admin_url( 'admin-ajax.php' ),
+			),
+			'extraUrlParams' => array(
+				'action'           => 'statify_track',
+				'_ajax_nonce'      => wp_create_nonce( 'statify_track' ),
+				'statify_referrer' => '${documentReferrer}',
+				'statify_target'   => '${canonicalPath}amp/',
+			),
+			'triggers'       => array(
+				'trackPageview' => array(
+					'on'      => 'visible',
+					'request' => 'pageview',
+				),
+			),
```

Comment:
As I wrote in Slack, I have no real experience with AMP, but one thing I noticed here: is it correct that the `transport` key was skipped from this config array?

Reply:
Good catch! Removing the section was not intentional.

Transport defaults to `beacon`, `xhrpost` and `image` enabled (in hierarchical order), so most browsers will use a compatible method by default. However `image` uses a GET request while _Statify_ requires POST, so I've disabled it explicitly.
Both `beacon` and `xhrpost` perform well.